qubes/ext/core_features.py

```diff
@@ -136,19 +138,22 @@ async def qubes_features_request(self, vm, event, untrusted_features):
             untrusted_feature_key,
             untrusted_feature_value,
         ) in untrusted_features.items():
-            if untrusted_feature_key.startswith("boot-mode.name."):
-                bootmode_key_parts = untrusted_feature_key.split(".")
-                if len(bootmode_key_parts) != 3:
-                    # Boot mode key contains unexpected data, reject it
-                    continue
-                bootmode_name = bootmode_key_parts[2]
-                if bootmode_name == "":
-                    continue
-                if (
-                    f"boot-mode.kernelopts.{bootmode_name}"
-                    not in new_bootmode_info
-                ) and bootmode_name != "default":
-                    continue
+            if not untrusted_feature_key.startswith("boot-mode."):
+                continue
+            bootmode_key_parts = untrusted_feature_key.split(".")
+            if len(bootmode_key_parts) != 3:
+                # Boot mode key contains unexpected data, reject it
+                continue
+            bootmode_name = bootmode_key_parts[2]
+            if bootmode_name == "":
+                continue
+            if (
+                f"boot-mode.kernelopts.{bootmode_name}" not in new_bootmode_info
+            ) and bootmode_name != "default":
+                continue
+            if untrusted_feature_key.startswith(
+                "boot-mode.name."
+            ) or untrusted_feature_key.startswith("boot-mode.default-user."):
```

Comment:
You allow setting `boot-mode.default-user.default`, but it's later ignored in `get_default_user()`. Please make it consistent (if you have some good argument for it, I can be convinced to allowing default user for the default boot mode too; otherwise don't allow setting it here).

Reply:
Fixed, test added for this edge case.